lib/shelf-metadata.js

```diff
@@ -0,0 +1,261 @@
+         */
+        constructor(uri, authToken) {
+            this.authToken = authToken;
+            this.data = "";
+            this.uri = uri;
+        }
+
+
+        /**
+         * Gets all of a file's metadata.
+         *
+         * @return {Promise.<Object>}
+         */
+        getAll() {
+            var options;
+
+            logger.debug("In ShelfMetadata.getAll()");
+            options = this.getOptions();
+            logger.info(`Getting metadata for ${options.url}`);
+            logger.debug(`GET request options: ${JSON.stringify(options, null, 4)}`);
+
+            return requestPromise.get(options).then((response) => {
+                logger.debug("In request.get().then()");
+
+                return utils.handleResponse(response);
+            });
+        }
+
+
+        /**
+         * Gets one metadata property from the file.
+         *
+         * @param {string} property The file's metadata property to get.
+         * @return {Promise.<Object>}
+         */
+        getProperty(property) {
+            var options;
+
+            logger.debug("In ShelfMetadata.getProperty()");
+            logger.debug(`Property param: ${property}`);
+
+            if (!property) {
+                throw new Error("Must provide a property to get.");
+            }
+
+            options = this.getOptions(property);
+            logger.info(`Getting metadata for ${options.url}`);
+            logger.debug(`GET request options: ${JSON.stringify(options, null, 4)}`);
+
+            return requestPromise.get(options).then((response) => {
+                logger.debug("In request.get().then()");
+
+                return utils.handleResponse(response);
+            });
+        }
+
+
+        /**
+         * Updates the file's metadata. Only updates mutable items.
+         *
+         * @param {Object} data The data to add to the file's metadata.
+         * @return {Promise.<Object>}
+         */
+        updateAll(data) {
+            var options;
+
+            logger.debug("In ShelfMetadata.updateAll()");
+            logger.debug(`Data param: ${data}`);
+
+            if (!data) {
+                throw new Error("Must provide data to update.");
+            }
+
+            options = this.getOptions();
+            options = this.setOptionsData(options, data);
+            logger.info(`Updating metadata for ${options.url}`);
+            logger.debug(`PUT request options: ${JSON.stringify(options, null, 4)}`);
+
+            return requestPromise.put(options).then((response) => {
+                logger.debug("In request.put().then()");
+
+                return utils.handleResponse(response);
+            });
+        }
+
+
+        /**
+         * Updates the file's metadata. Only updates mutable items.
+         * Due to the nature of the Shelf API, this will create the property
+         * if it doesn't exist.
+         *
+         * @param {string} property The metadata property to update.
```

Comment:
Yeah, I usually try and name things better. Since the metadata can be pretty arbitrary, I felt like it was okay to name it as such. I will try to come up with something better.